examples/energy-management-app/linux/main.cpp

```diff
@@ -87,22 +99,56 @@ static uint32_t ParseNumber(const char * pString)
 void ApplicationInit()
 {
     ChipLogDetail(AppServer, "Energy Management App: ApplicationInit()");
-    EvseApplicationInit();
+    if (strcmp(spApp, kEvseApp) == 0)
+    {
+        EvseApplicationInit();
+    }
+    else if (strcmp(spApp, kWhmApp) == 0)
+    {
+        FullWhmApplicationInit();
+    }
+    else
+    {
+        ChipLogError(Support, "Unexpected application %s", spApp);
+    }
 }
 
 void ApplicationShutdown()
 {
     ChipLogDetail(AppServer, "Energy Management App: ApplicationShutdown()");
+
     EvseApplicationShutdown();
+    FullWhmApplicationShutdown();
```

Comment:
I see here we might shutdown an app that hasn't been initialized, is there no risk of undefined behavior here?

Reply:
I think all of the functions check that they only free pointers which have been allocated. So even if you run the app in EVSE mode, it is safe to call the WaterHeaterShutdown.